src/components/collapsible-panel.tsx

```diff
@@ -0,0 +1,43 @@
+import { ToolbarButtonComponent } from '@jupyterlab/apputils';
+import React, { useState } from 'react';
+import { caretDownIcon, caretRightIcon } from '@jupyterlab/ui-components';
+
+export interface ICollapsiblePanelProps {
+  title: JSX.Element | string;
+  content: JSX.Element | string;
+  expanded?: boolean;
+}
+
+export function CollapsiblePanel(props: ICollapsiblePanelProps): JSX.Element {
+  const [expanded, setExpanded] = useState(props.expanded ?? false);
+
+  const handleChange = () => {
+    setExpanded(!expanded);
```

Comment:
Is this function used anywhere else? If not, it's probably easier to include its one line in the `onClick` argument's value.

Reply:
Its used on twice / on 2 elements: summary icon and summary title. Based on this decided to put click logic into single function as a single source of truth

Reply:
After simplifying the number of containers used, changed to inline function on `onClick` as suggested